File: BFS_DFS/1260.py
# [BOJ-Silver2] 1260_DFS와 BFS

from collections import deque

n, m, v = map(int, input().split())
graph = [[] for _ in range(n + 1)]
for _ in range(m):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)
    
# 정점 번호가 작은 순으로 방문
for i in graph:
    i.sort()
    
visited1 = [False] * (n+1)
visited2 = [False] * (n+1) 
    
# BFS    
def bfs(v):
   visited1[v] = True    
   queue = deque([v])
   
   while queue:
       x = queue.popleft()
       print(x, end=' ')
       
       for i in graph[x]:
            if not visited1[i]:
                visited1[i] = True
                queue.append(i)
                

# DFS
def dfs(v):
    visited2[v] = True
    print(v, end=' ')
    
    for i in graph[v]:
        if not visited2[i]:
            dfs(i)

dfs(v)
print()
bfs(v)

File: BFS_DFS/_1260.java
package BFS_DFS;
// [BOJ-Sliver2] 1260_DFS와 BFS

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class _1260 {
	static int n, m, v;
	static boolean[] visited;
	static boolean[][] graph;

	static StringBuilder result = new StringBuilder();

	public static void dfs(int start) {

		for (int i = 1; i <= n; i++) {
			if (graph[start][i] && !visited[i]) {
				visited[i] = true;
				result.append(i).append(" ");
				dfs(i);
			}
		}
	}

	public static void bfs(int start) {
		Queue<Integer> queue = new LinkedList<>();
		queue.add(start);
		visited[start] = true;

		while (!queue.isEmpty()) {
			int curr = queue.poll();
			result.append(curr).append(" ");
			for (int i = 1; i <= n; i++) {
				if (graph[curr][i] && !visited[i]) {
					visited[i] = true;
					queue.add(i);
				}
			}
		}
	}

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		n = Integer.parseInt(st.nextToken());
		m = Integer.parseInt(st.nextToken());
		v = Integer.parseInt(st.nextToken());

		graph = new boolean[n + 1][n + 1];
		visited = new boolean[n + 1];

		for (int i = 0; i < m; i++) {
			st = new StringTokenizer(br.readLine());
			int a = Integer.parseInt(st.nextToken());
			int b = Integer.parseInt(st.nextToken());
			graph[a][b] = true;
			graph[b][a] = true;
		}

		visited[v] = true;
		result.append(v).append(" ");
		dfs(v);
		result.append('\n');

		Arrays.fill(visited, false);
		bfs(v);

		System.out.println(result);

	}
}
